connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/PostgresIntegrationSuite.scala

```diff
@@ -379,4 +386,32 @@ class PostgresIntegrationSuite extends DockerJDBCIntegrationSuite {
     assert(row(0).length === 1)
     assert(row(0).getString(0) === "$1,000.00")
   }
+
+  test("SPARK-43040: timestamp_ntz read test") {
+    val prop = new Properties
+    prop.setProperty("preferTimestampNTZ", "true")
+    val df = sqlContext.read.jdbc(jdbcUrl, "timestamp_ntz", prop)
+    val row = df.collect()
```

Comment:
do we enable java8 datetime API somewhere? otherwise `df.collect` should return `java.sql.Timestamp`

Reply:
Do you mean `spark.sql.datetime.java8API.enabled`? This only controls `TimestampType` but not `TimestampNTZType` if I am not mistaken.
`TimestampNTZType` always converts to `LocalDateTime`.

Reply:
ah I see